src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -58,25 +58,20 @@ private WebAssemblyHostBuilder()
 
             InitializeDefaultServices();
 
+            var hostEnvironment = InitializeEnvironment(jsRuntime);
+
             _createServiceProvider = () =>
             {
-                return Services.BuildServiceProvider();
+                return Services.BuildServiceProvider(validateScopes: hostEnvironment.Environment == "Development");
```

Comment:
This is the actual code change. The remainder of this is refactoring to allow unit testing this.